packages/flutter_tools/lib/src/localizations/gen_l10n_types.dart

```diff
@@ -309,48 +313,96 @@ class Placeholder {
 // The value of this Message is "Hello World". The Message's value is the
 // localized string to be shown for the template ARB file's locale.
 // The docs for the Placeholder explain how placeholder entries are defined.
-// TODO(thkim1011): We need to refactor this Message class to own all the messages in each language.
-// See https://github.com/flutter/flutter/issues/112709.
 class Message {
-  Message(Map<String, Object?> bundle, this.resourceId, bool isResourceAttributeRequired)
-    : assert(bundle != null),
+  Message(
+    AppResourceBundle templateBundle,
+    AppResourceBundleCollection allBundles,
+    this.resourceId,
+    bool isResourceAttributeRequired
+  )
```

Comment:
Formatting is unconventional here. Per https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#constructor-syntax it should be: 
```dart
  Message(
    AppResourceBundle templateBundle,
    AppResourceBundleCollection allBundles,
    this.resourceId,
    bool isResourceAttributeRequired
  ) : assert(templateBundle != null),
      assert(allBundles != null),
      assert(resourceId != null && resourceId.isNotEmpty),
      ....
```

Reply:
Fixed!